File: robot-server/src/main/java/com/informatica/servidor/logica/Mano.java
package com.informatica.servidor.logica;

import lombok.Getter;

/**
 *
 * @author Vamaya
 */
public class Mano extends HuesoConHueso{
    
    //VARIABLES    
    @Getter private final Dedo dedoDer;
    
    //CONSTRUCTOR
    public Mano(Dedo dedoIzq, Dedo dedoDer, int xFinal, int yFinal, int xInicial, int yInicial, int ancho) {
        super(dedoIzq, xFinal, yFinal, xInicial, yInicial, ancho);
        this.dedoDer = dedoDer;
    }
    
    //METODOS
    @Override
    public void girarHueso(int alphaGrados){
        super.girarHueso(alphaGrados);
        int xn = this.getXFinalDinamica();
        int yn= this.getYFinalDinamica();
        dedoDer.cambiarCoordenadasBase(xn, yn);
    }
    
    @Override
    public void cambiarCoordenadasBase(int xTrasladada, int yTrasladada){
        super.cambiarCoordenadasBase(xTrasladada, yTrasladada);
        int xnf = this.getXFinalDinamica();
        int ynf= this.getYFinalDinamica();
        dedoDer.cambiarCoordenadasBase(xnf, ynf);
    }

}

File: my-app/src/logica/Mano.ts
import { Dedo } from './Dedo';
import { HuesoConHueso } from './HuesoConHueso';

export class Mano extends HuesoConHueso {

    dedoDer: Dedo;

    constructor (dedoIzq: Dedo,  dedoDer: Dedo,  xFinal: number,  yFinal: number,  xInicial: number,  yInicial: number,  ancho: number) {
        super(dedoIzq, xFinal, yFinal, xInicial, yInicial, ancho);
        this.dedoDer = dedoDer;
    }

    //METODOS

    girarHueso(alphaGrados: number){
        super.girarHueso(alphaGrados);
        let  xn: number = this.xFinalDinamica;
        let  yn: number= this.yFinalDinamica;
        this.dedoDer.cambiarCoordenadasBase(xn, yn);
    }


    cambiarCoordenadasBase(xTrasladada: number, yTrasladada: number){
        super.cambiarCoordenadasBase(xTrasladada, yTrasladada);
        let  xnf: number = this.xFinalDinamica;
        let  ynf: number = this.yFinalDinamica;
        this.dedoDer.cambiarCoordenadasBase(xnf, ynf);
    }
}
